applications/luci-app-aria2/root/usr/libexec/aria2-call

```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
+
+. "$IPKG_INSTROOT/usr/share/libubox/jshn.sh"
+
+action=$1
+shift
+
+case "$action" in
+	cat)
+		config_dir="$(uci -q get aria2.main.config_dir)"
+		list_file="${config_dir:-/var/etc/aria2}/aria2.$1.main"
```

Comment:
This is almost golden! :grinning:  But we need to validate `$1` yet in order to prevent path traversal attempts (which likely fail due to the forced `aria2.` prefix and `.main` suffix, but still.

Wrap the above into something like:

```bash
cat)
	case "$1" in
	conf|session)
		config_dir="$(uci -q get aria2.main.config_dir)"
		list_file="${config_dir:-/var/etc/aria2}/aria2.$1.main"
		json_init
		json_add_string file "$list_file"
		json_add_string content "$(cat "$list_file")"
		json_dump
	;;
	*)
		# maybe emit an error here or simply do nothing (in this case, drop the entire "*) ... ;;" case)
	;;
;;
```

Reply:
> Wrap the above into something like:

OK

The CI build has failed for some reasons which seems have nothing to do with this LuCI app.